tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -154,6 +157,139 @@ public static void fastStartSpan(String operation) {
         fastStartSpan(operation, SpanType.LOCAL);
     }
 
+    /**
+     * Like {@link #startSpan(String, SpanType)}, but does not set or modify tracing thread state.
+     * This is an internal utility that should not be called directly outside of {@link DetachedSpan}.
+     */
+    static DetachedSpan detachInternal(String operation, SpanType type) {
+        Trace maybeCurrentTrace = currentTrace.get();
+        String traceId = maybeCurrentTrace != null
+                ? maybeCurrentTrace.getTraceId() : Tracers.randomId();
+        boolean sampled = maybeCurrentTrace != null
+                ? maybeCurrentTrace.isObservable() : sampler.sample();
+        return sampled
+                ? new SampledDetachedSpan(operation, type, traceId, getParentSpanId(maybeCurrentTrace))
+                : new UnsampledDetachedSpan(traceId);
+    }
+
+    private static Optional<String> getParentSpanId(@Nullable Trace trace) {
+        if (trace != null) {
+            Optional<OpenSpan> maybeOpenSpan = trace.top();
+            if (maybeOpenSpan.isPresent()) {
+                return Optional.of(maybeOpenSpan.get().getSpanId());
+            }
+        }
+        return Optional.empty();
+    }
+
+    private static final class SampledDetachedSpan implements DetachedSpan {
+
+        private final AtomicBoolean completed = new AtomicBoolean();
+        private final String traceId;
+        private final OpenSpan openSpan;
+
+        SampledDetachedSpan(
+                String operation, SpanType type, String traceId, Optional<String> parentSpanId) {
+            this.traceId = traceId;
+            this.openSpan = OpenSpan.builder()
+                    .parentSpanId(parentSpanId)
+                    .spanId(Tracers.randomId())
+                    .operation(operation)
+                    .type(type)
+                    .build();
+        }
+
+        @Override
+        public SpanToken attach(String operationName, SpanType type) {
+            warnIfCompleted("startSpanOnCurrentThread");
+            Trace maybeCurrentTrace = currentTrace.get();
+            setTrace(Trace.of(true, traceId));
+            Tracer.fastStartSpan(operationName, openSpan.getSpanId(), type);
+            return TraceRestoringSpanToken.of(maybeCurrentTrace);
+        }
+
+        @Override
+        public DetachedSpan detach(String operation, SpanType type) {
```

Comment:
What use case do you envision for this?

Reply:
Consider a webserver: When a request is received on a non-blocking thread, I'd like to start a detached span representing the entire request. Later, before processing, the request is enqueued to a larger thread pool where we allow work -- I'd like the ability to track time spent enqueued, despite not having attached the span to thread.